aws/resource_aws_prometheus_workspace_test.go

```diff
@@ -0,0 +1,121 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/prometheusservice"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAWSAMPWorkspace_basic(t *testing.T) {
+	workspaceAlias := acctest.RandomWithPrefix("tf_amp_workspace")
+	resourceName := "aws_prometheus_workspace.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAMPWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAWSAMPWorkspaceConfigWithAlias(workspaceAlias),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAWSAMPWorkspaceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "alias", workspaceAlias),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAWSAMPWorkspaceConfigWithoutAlias(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "alias", ""),
+				),
+			},
+			{
+				Config: testAWSAMPWorkspaceConfigWithAlias(workspaceAlias),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAWSAMPWorkspaceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "alias", workspaceAlias),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAWSAMPWorkspaceExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No AMP Workspace ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).prometheusconn
+
+		req := &prometheusservice.DescribeWorkspaceInput{
+			WorkspaceId: aws.String(rs.Primary.ID),
+		}
+		describe, err := conn.DescribeWorkspace(req)
+		if err != nil {
+			return err
+		}
+		if describe == nil {
+			return fmt.Errorf("Got nil account ?!")
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAMPWorkspaceDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).prometheusconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_prometheus_workspace" {
+			continue
+		}
+
+		res, err := conn.DescribeWorkspace(&prometheusservice.DescribeWorkspaceInput{
+			WorkspaceId: aws.String(rs.Primary.ID),
+		})
+		if err == nil {
+			if aws.StringValue(res.Workspace.Status.StatusCode) != "DELETING" {
```

Comment:
This typically indicates that the resource should be implementing waiter functions for creation, deletion, etc. Some examples can be see with other resources that implement `aws/internal/service/*/waiter` packages. It is on my plate to add this information into the contributing guide.

Reply:
I've added waiters in the next rev.